buildtest/buildsystem/base.py

```diff
@@ -1,8 +1,8 @@
 """
```

Comment:
One quick thought - why not just do BuildParser?

Reply:
I think ``BuildspecParser`` is more specific to say it's parsing and validating a ``Buildspec`` as pose to ``BuildParser``. If there are other objects other than ``Buildspec`` that buildtest supports then ``BuildParser`` would be too vague. It won't tell what it is parsing

Reply:
Yes, I think that's right, I'm okay with leaving the name as is, even though it's a little long :) But what other kind of Parser do you anticipate?

Reply:
What i had in mind, though i dont have this thought out correctly is the following...

At the moment we are talking about a ``Buildspec`` that is a YAML file that is being parsed into a test script. At the moment buildtest does not have a concept of objects similar to ``kind`` in Kubernetes. In testing realm there could be objects such as ``Makefiles``, ``Dockerfile``, ``Singularityfile``.  A Buildspec that defines such an object would be parsed in a different way, or even delegated to a subclass of ``BuildspecParser`` such that it can validate the buildspec along with generated content such as Dockerfile, Singularityfile. Again these are bit out of scope and I dont have a clear picture how this will work out but certainly i think it adds value when user can specify what type of test object to generate. buildtest would need to parse correctly hence ``BuildParser`` would be bit too vague. 

There would be classes for each type of parser to ensure each artifact generated by buildtest is valid before executing.  I think it's bit hard to validate a test script or a job script without executing it. 
I think some basic validation for a test script would be something like

- check if shebang is present ``#!/bin/bash``
- check if test has read and execute permission (minimum ``555``)
- check if test has ``.sh``, ``.bash`` extension 

For jobscript it can be as simple as checking if there is atleast one scheduler directives in ``#SBATCH`` in the  file.

Again all of this is out of scope but something to think about

Reply:
Oh that would make sense - if there is some other config file that warrants a special parsing. +1.